File: Python/Day 1/task2.py
file = open("C:\\Source\\amortsell\\AdventOfCode2021\\Input\\Day 1.txt", "r")
lines = file.readlines()

def arrSum(arr):
  sum = 0
  for i in arr:
    sum += i
  return sum 

count = 0
arr1 = []
arr2 = []
arr3 = [] 
wait = 2
for line in lines:
  if wait >= 0:
    arr1.append(int(line))
  if wait < 2:
    arr2.append(int(line))
  if wait < 1:
    arr3.append(int(line))
  if wait < 0:
    if arrSum(arr2) > arrSum(arr1):
      count += 1
    arr1 = arr2.copy()
    arr2 = arr3.copy()
    arr3.clear()
    arr3.append(int(line))
  wait -= 1    

print(count)

File: Typescript/Day 1/task2.ts
/// <reference path="../../node_modules/@types\\node/fs.d.ts" />

import * as fs  from 'fs';

let lines: string[] = []; 
const text = fs.readFileSync('Input/Day 1.txt', 'utf8').toString();
lines.push(...text.split('\n'));

let count = 0;
lines.map((x, i, arr) => {
  count += i > 2 ? arr.slice(i - 2, i + 1).reduce((prev, n) => parseInt(n) + prev, 0) > arr.slice(i-3, i).reduce((prev, n) => parseInt(n) + prev, 0) ? 1 : 0: 0;
});
console.log(count);